core/domain/classifier_services.py

```diff
@@ -100,8 +101,39 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id)
+    print exploration.title
+    state_obj = exploration.states[state_name]
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput']
+    if state_obj.classifier_model_id:
+        classifier = get_classifier_by_id(state_obj.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            algorithm_id)
+        sc.from_dict(sc_dict)
+
+    else:
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            algorithm_id)
         sc.train(training_examples)
+        cached_classifier_data = sc.to_dict()
+        schema_version = exploration.states_schema_version
```

Comment:
No @pranavsid98, this is not the same schema version. We will have to create a mapping of algorithm to schema versions in feconf and use that schema version here. I think we had discussed about this already.